tool/src/org/antlr/v4/codegen/model/ListenerFile.java

```diff
@@ -61,7 +61,8 @@ public ListenerFile(OutputModelFactory factory, String fileName) {
 			}
 		}
 		ActionAST ast = g.namedActions.get("header");
-		if ( ast!=null ) header = new Action(factory, ast);
+		if ( ast!=null && ast.getScope() == null)
```

Comment:
just a reminder that I'd like { } around even single statements :)

Reply:
;-) there wasn't one originally

Reply:
damn...has eroded. Maybe the rule is if it's one line like that but not two lines.